src/Simulator/FireSimulator.cpp

```diff
@@ -97,20 +99,39 @@ void cFireSimulator::SimulateChunk(std::chrono::milliseconds a_Dt, int a_ChunkX,
 		int x = itr->x;
 		int y = itr->y;
 		int z = itr->z;
+		auto AbsPos = cChunkDef::RelativeToAbsolute({x, y, z}, a_Chunk->GetPosX(), a_Chunk->GetPosZ());
 		BLOCKTYPE BlockType = a_Chunk->GetBlock(x, y, z);
 
 		if (!IsAllowedBlock(BlockType))
 		{
 			// The block is no longer eligible (not a fire block anymore; a player probably placed a block over the fire)
 			FLOG("FS: Removing block {%d, %d, %d}",
-				itr->x + a_ChunkX * cChunkDef::Width, itr->y, itr->z + a_ChunkZ * cChunkDef::Width
+				AbsPos.x, AbsPos.y, AbsPos.z
 			);
 			itr = Data.erase(itr);
 			continue;
 		}
 
+		auto BurnsForever = ((y < 0) || DoesBurnForever(a_Chunk->GetBlock(x, (y - 1), z)));
```

Comment:
This isn't quite right:
 * The check should be `y <= 0` because `y - 1` should be a valid block coord.
 * If the fire is at `y <= 0` then it _shouldn't_ burn forever because there is no block below which burns forever.

Reply:
Looking at my comment again it's a bit confusing so here's the code I had in mind:
```cpp
auto BurnsForever = (
    (y > 0) &&  // Must have a block below it
    DoesBurnForever(a_Chunk->GetBlock(x, (y - 1), z))  // And that block does burn forever
);
```

Reply:
If you look again at the later commits, I realised that and fixed it :P